libr/bin/dwarf.c

```diff
@@ -2447,6 +2460,11 @@ static HtUP *parse_loc_raw(HtUP/*<offset, List *<LocListEntry>*/ *loc_table, con
 			range = NULL;
 		}
 	}
+	/* if for some reason end of list is missing, then loc_list would leak */
+	if (loc_list) { /* looks ugly, recycling existing function so I don't dupe code */
```

Comment:
instead of this ugliness + comment of ugliness you can do:
```C
static void free_loc_table_list(RBinDwarfLocList *loc_list) {
	RListIter *iter;
	RBinDwarfLocRange *range;
	r_list_foreach (loc_list->list, iter, range) {
		free (range->expression->data);
		free (range->expression);
		free (range);
	}
	r_list_free (loc_list->list);
	free (loc_list);
}

static void free_loc_table_entry(HtUPKv *kv) {
	if (kv) {
		free_loc_table_list (kv->value);
	}
}
```

and here just do:
```C
if (loc_list) {
	free_loc_table_list (loc_list);
}
```

Reply:
Yes, that's better